testing/mm: Test timeout while running TestDeposit/not_withdrawer,_dynamic_swapper.

On #3061 so close to master, runnig the `run_tests.sh` script hit the 10 minute timeout, which probably indicates a deadlock but may be only in tests:

<details closed>
<summary>panic log</summary>

```
ok      decred.org/dcrdex/client/db/bolt        1.671s
ok      decred.org/dcrdex/client/db/test        2.380s
2024-11-08 16:07:05.674 [INF] mm_TEST: Deposited 0.50000000 DCR. TxID = 
2024-11-08 16:07:05.675 [INF] mm_TEST: Withdrew 0.50305000 DCR
2024-11-08 16:07:05.675 [INF] mm_TEST: Deposited 0.05688000 BTC. TxID = 
2024-11-08 16:07:05.675 [INF] mm_TEST: Withdrew 0.04955500 BTC
2024-11-08 16:07:05.677 [INF] mm_TEST: Deposited 0.50000000 DCR. TxID = 
2024-11-08 16:07:05.677 [INF] mm_TEST: Withdrew 0.50305000 DCR
2024-11-08 16:07:05.677 [INF] mm_TEST: Deposited 0.005688000 ETH. TxID = 
2024-11-08 16:07:05.678 [INF] mm_TEST: Withdrew 0.004949500 ETH
2024-11-08 16:07:05.679 [INF] mm_TEST: Deposited 0.050000000 ETH. TxID = 
2024-11-08 16:07:05.680 [INF] mm_TEST: Withdrew 0.050300000 ETH
2024-11-08 16:07:05.680 [INF] mm_TEST: Deposited 0.05688000 DCR. TxID = 
2024-11-08 16:07:05.680 [INF] mm_TEST: Withdrew 0.04955500 DCR
2024-11-08 16:07:05.682 [INF] mm_TEST: Deposited 50.000000 USDC. TxID = 
2024-11-08 16:07:05.682 [INF] mm_TEST: Withdrew 50.000000 USDC
2024-11-08 16:07:05.683 [INF] mm_TEST: Deposited 0.005655050 ETH. TxID = 
2024-11-08 16:07:05.683 [INF] mm_TEST: Withdrew 0.005060500 ETH
2024-11-08 16:07:05.684 [INF] mm_TEST: Deposited 0.050000000 ETH. TxID = 
2024-11-08 16:07:05.684 [INF] mm_TEST: Withdrew 0.050410000 ETH
2024-11-08 16:07:05.685 [INF] mm_TEST: Deposited 5.588100 USDC. TxID = 
2024-11-08 16:07:05.685 [INF] mm_TEST: Withdrew 4.949500 USDC
2024-11-08 16:07:05.687 [INF] mm_TEST: Deposited 0.050000000 WETH. TxID = 
2024-11-08 16:07:05.687 [INF] mm_TEST: Withdrew 0.050000000 WETH
2024-11-08 16:07:05.687 [INF] mm_TEST: Deposited 5.588100 USDC. TxID = 
2024-11-08 16:07:05.688 [INF] mm_TEST: Withdrew 4.949500 USDC
2024-11-08 16:07:05.690 [INF] mm_TEST: Deposited 50.000000 USDC. TxID = 
2024-11-08 16:07:05.690 [INF] mm_TEST: Withdrew 50.000000 USDC
2024-11-08 16:07:05.691 [INF] mm_TEST: Deposited 0.005655050 WETH. TxID = 
2024-11-08 16:07:05.691 [INF] mm_TEST: Withdrew 0.004930500 WETH
2024-11-08 16:07:05.694 [INF] mm_TEST: Deposited 50.000000 USDC. TxID = 
2024-11-08 16:07:05.694 [INF] mm_TEST: Withdrew 50.000000 USDC
2024-11-08 16:07:05.695 [INF] mm_TEST: Deposited 0.05565050 WBTC. TxID = 
2024-11-08 16:07:05.695 [INF] mm_TEST: Withdrew 0.04948500 WBTC
2024-11-08 16:07:05.698 [INF] mm_TEST: Deposited 0.50000000 WBTC. TxID = 
2024-11-08 16:07:05.698 [INF] mm_TEST: Withdrew 0.50000000 WBTC
2024-11-08 16:07:05.698 [INF] mm_TEST: Deposited 5.558400 USDC. TxID = 
2024-11-08 16:07:05.699 [INF] mm_TEST: Withdrew 4.950400 USDC
2024-11-08 16:07:05.701 [INF] mm_TEST: Deposited 0.50000000 DCR. TxID = 
2024-11-08 16:07:05.701 [INF] mm_TEST: Withdrew 0.50305000 DCR
2024-11-08 16:07:05.702 [INF] mm_TEST: Deposited 0.05598000 WBTC. TxID = 
2024-11-08 16:07:05.702 [INF] mm_TEST: Withdrew 0.04947500 WBTC
2024-11-08 16:07:05.705 [INF] mm_TEST: Deposited 0.50000000 WBTC. TxID = 
2024-11-08 16:07:05.705 [INF] mm_TEST: Withdrew 0.50000000 WBTC
2024-11-08 16:07:05.705 [INF] mm_TEST: Deposited 0.05658300 DCR. TxID = 
2024-11-08 16:07:05.706 [INF] mm_TEST: Withdrew 0.04956400 DCR
2024-11-08 16:07:05.722 [WRN] mm_TEST: multiTrade: rate 20000000 causes self match. Placements should be farther from mid-gap.
2024-11-08 16:07:05.722 [WRN] mm_TEST: multiTrade: rate 20000000 causes self match. Placements should be farther from mid-gap.
2024-11-08 16:07:05.722 [WRN] mm_TEST: multiTrade: rate 20000000 causes self match. Placements should be farther from mid-gap.
2024-11-08 16:07:05.723 [WRN] mm_TEST: multiTrade: rate 30000000 causes self match. Placements should be farther from mid-gap.
2024-11-08 16:07:05.723 [WRN] mm_TEST: multiTrade: rate 30000000 causes self match. Placements should be farther from mid-gap.
2024-11-08 16:07:05.724 [WRN] mm_TEST: multiTrade: rate 30000000 causes self match. Placements should be farther from mid-gap.
2024-11-08 16:07:05.746 [INF] mm_TEST: Deposited 0.01000000 DCR. TxID = 1529c99086b77259ee8bd66dcbeb2f2cdc1d36abf883acc87f2a6c73e9545d95
2024-11-08 16:07:05.849 [INF] mm_TEST: Deposited 0.01000000 DCR. TxID = 1529c99086b77259ee8bd66dcbeb2f2cdc1d36abf883acc87f2a6c73e9545d95
2024-11-08 16:07:05.950 [INF] mm_TEST: Deposited 0.01000000 DCR. TxID = 1529c99086b77259ee8bd66dcbeb2f2cdc1d36abf883acc87f2a6c73e9545d95
panic: test timed out after 10m0s
        running tests:
                TestDeposit (10m0s)
                TestDeposit/not_withdrawer,_dynamic_swapper (9m59s)

goroutine 338 [running]:
testing.(*M).startAlarm.func1()
        /usr/lib/go/src/testing/testing.go:2373 +0x265
created by time.goFunc
        /usr/lib/go/src/time/sleep.go:215 +0x45

goroutine 1 [chan receive, 9 minutes]:
testing.(*T).Run(0xc0000f4680, {0x2635807, 0xb}, 0x28b5010)
        /usr/lib/go/src/testing/testing.go:1751 +0x851
testing.runTests.func1(0xc0000f4680)
        /usr/lib/go/src/testing/testing.go:2168 +0x86
testing.tRunner(0xc0000f4680, 0xc0005cfae0)
        /usr/lib/go/src/testing/testing.go:1690 +0x227
testing.runTests(0xc000013278, {0x37b7ec0, 0x19, 0x19}, {0x7a887ca15a68?, 0x40?, 0x3845880?})
        /usr/lib/go/src/testing/testing.go:2166 +0x8bf
testing.(*M).Run(0xc0001148c0)
        /usr/lib/go/src/testing/testing.go:2034 +0xf18
main.main()
        _testmain.go:93 +0x165

goroutine 51 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 52 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 53 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 54 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 55 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 56 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 57 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 58 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 59 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 60 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 61 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 62 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 63 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 64 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 65 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 66 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 67 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 68 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 69 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 70 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 71 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 72 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 73 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 74 [chan receive, 10 minutes]:
github.com/ethereum/go-ethereum/core.(*txSenderCacher).cache(0xc0003842a0)
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:63 +0x68
created by github.com/ethereum/go-ethereum/core.newTxSenderCacher in goroutine 1
        /home/joe/go/pkg/mod/github.com/ethereum/go-ethereum@v1.14.8/core/sender_cacher.go:55 +0xab

goroutine 298 [chan receive, 9 minutes]:
testing.(*T).Run(0xc00030c820, {0x2660717, 0x1f}, 0xc0003d25a0)
        /usr/lib/go/src/testing/testing.go:1751 +0x851
decred.org/dcrdex/client/mm.TestDeposit.func1(...)
        /home/joe/git/dcrdex/client/mm/exchange_adaptor_test.go:3962
decred.org/dcrdex/client/mm.TestDeposit(0xc00030c820)
        /home/joe/git/dcrdex/client/mm/exchange_adaptor_test.go:4096 +0x33bc
testing.tRunner(0xc00030c820, 0x28b5010)
        /usr/lib/go/src/testing/testing.go:1690 +0x227
created by testing.(*T).Run in goroutine 1
        /usr/lib/go/src/testing/testing.go:1743 +0x826

goroutine 312 [sync.RWMutex.Lock, 9 minutes]:
sync.runtime_SemacquireRWMutex(0xc0004e7d94?, 0x1?, 0xc00026a7e0?)
        /usr/lib/go/src/runtime/sema.go:105 +0x25
sync.(*RWMutex).Lock(0xc0004e7d80)
        /usr/lib/go/src/sync/rwmutex.go:153 +0x8b
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).confirmDeposit(0xc0004e2600, {0x2b782a8, 0xc0004e7c20}, {0xc000419780, 0x40})
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:1724 +0x389
decred.org/dcrdex/client/mm.TestDeposit.TestDeposit.func1.func2(0xc00030cd00)
        /home/joe/git/dcrdex/client/mm/exchange_adaptor_test.go:4049 +0x177c
testing.tRunner(0xc00030cd00, 0xc0003d25a0)
        /usr/lib/go/src/testing/testing.go:1690 +0x227
created by testing.(*T).Run in goroutine 298
        /usr/lib/go/src/testing/testing.go:1743 +0x826

goroutine 313 [chan receive, 9 minutes]:
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect.func1()
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3332 +0xd1
created by decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect in goroutine 312
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3330 +0x49d

goroutine 314 [chan receive, 9 minutes]:
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect.func2()
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3339 +0xa5
created by decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect in goroutine 312
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3337 +0x572

goroutine 315 [select, 9 minutes]:
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect.func3()
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3351 +0x265
created by decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect in goroutine 312
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3345 +0x645

goroutine 316 [select, 9 minutes]:
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect.func4()
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3365 +0x189
created by decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).Connect in goroutine 312
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:3361 +0x714

goroutine 317 [chan receive, 9 minutes]:
decred.org/dcrdex/client/mm.dummyLooper.func1()
        /home/joe/git/dcrdex/client/mm/mm_arb_market_maker_test.go:494 +0x4a
created by decred.org/dcrdex/client/mm.dummyLooper in goroutine 312
        /home/joe/git/dcrdex/client/mm/mm_arb_market_maker_test.go:493 +0x111

goroutine 318 [semacquire, 9 minutes]:
sync.runtime_Semacquire(0xc00070e778?)
        /usr/lib/go/src/runtime/sema.go:71 +0x25
sync.(*WaitGroup).Wait(0xc00070e770)
        /usr/lib/go/src/sync/waitgroup.go:118 +0xa5
decred.org/dcrdex/dex.(*ConnectionMaster).Connect.func1()
        /home/joe/git/dcrdex/dex/runner.go:140 +0x3d
created by decred.org/dcrdex/dex.(*ConnectionMaster).Connect in goroutine 312
        /home/joe/git/dcrdex/dex/runner.go:139 +0x2c5

goroutine 319 [sync.RWMutex.RLock, 9 minutes]:
sync.runtime_SemacquireRWMutexR(0xc0004e7d90?, 0x1?, 0x485a59?)
        /usr/lib/go/src/runtime/sema.go:100 +0x25
sync.(*RWMutex).RLock(0xc0004e7d80)
        /usr/lib/go/src/sync/rwmutex.go:72 +0x65
decred.org/dcrdex/client/mm.(*pendingDeposit).balanceEffects(0xc0004e7d60)
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:237 +0x5c
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).updateDepositEvent(0xc0004e2600, 0xc0004e7d60)
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:763 +0x65
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).confirmDeposit.func1()
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:1709 +0x56
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).confirmDeposit(0xc0004e2600, {0x2b782a8, 0xc0004e7c20}, {0xc000419780, 0x40})
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:1756 +0x6b5
decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).deposit.func1()
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:1819 +0x272
created by decred.org/dcrdex/client/mm.(*unifiedExchangeAdaptor).deposit in goroutine 312
        /home/joe/git/dcrdex/client/mm/exchange_adaptor.go:1812 +0xa5a
FAIL    decred.org/dcrdex/client/mm     600.132s
ok      decred.org/dcrdex/client/mm/libxc       1.047s
```
</details>

--- a/client/mm/exchange_adaptor.go
+++ b/client/mm/exchange_adaptor.go
@@ -760,7 +760,7 @@ func (u *unifiedExchangeAdaptor) updateDepositEvent(deposit *pendingDeposit) {
 	e := &MarketMakingEvent{
 		ID:             deposit.eventLogID,
 		TimeStamp:      deposit.timestamp,
-		BalanceEffects: combineBalanceEffects(deposit.balanceEffects()),
+		BalanceEffects: combineBalanceEffects(depositBalanceEffects(deposit.assetID, deposit.tx, deposit.cexConfirmed)),
 		Pending:        !deposit.cexConfirmed || !deposit.feeConfirmed,
 		DepositEvent: &DepositEvent{
 			AssetID:     deposit.assetID,

